modules/videopress/shortcode.php

```diff
@@ -36,6 +38,33 @@ public static function initialize() {
 		return self::$instance;
 	}
 
+	/**
+	 * Filters old <embed> codes out of the content
+	 *
+	 * @param string $content the content.
+	 * @return string filtered content
+	 */
+	public function reverse_old_embeds( $content ) {
+		$regex   = '%<embed[^>]*+>(?:\s*</embed>)?%i';
+		$content = preg_replace_callback(
+			$regex,
+			function( $matches, $orig_html = null ) {
+				$embed_code  = $matches[0];
+				$url_matches = array();
+
+				// get video ID from flash URL.
+				$url_matched = preg_match( '/src="http:\/\/v.wordpress.com\/([^"]+)"/', $embed_code, $url_matches );
+
+				if ( $url_matched ) {
+					$video_id = $url_matches[1];
+					return "[videopress $video_id]";
```

Comment:
Instead of transforming into shortcodes, maybe we could transform into full URLs instead, and let oEmbed take care of displaying the video player?

Reply:
I agree with this. I'd want to test it out without the shortcodes module enabled. The code as is, I think, would then just output a raw shortcode, which is an equally broken UX, so the oembed (which should be supported by Core) is a more universal solution.

Reply:
> I'd want to test it out without the shortcodes module enabled.

This shouldn't be an issue as long as they have the VideoPress module enabled. The module force-loads the VideoPress shortcode:
https://github.com/Automattic/jetpack/blob/593ee4cccd6aead047c72aa78378f368f084035e/modules/videopress.php#L15

Reply:
Do we want to limit it to those with the VideoPress module? Someone embeds a VideoPress on their Jetpack site, whether or not they had VideoPress themselves, or they previously had VideoPress and downgraded. Those instances wouldn't have the module active.

Reply:
That's a good point. Do you think we should filter `the_content` as soon as Jetpack is connected to WordPress.com, or even active? I'm personally okay with limiting things to sites that use either the Shortcodes module or the VideoPress module, but I'd be okay doing this for everyone.

Reply:
I could be convinced either way. Since the old embed is totally broken, I would want to fix it for as many people as possible to preserve a good visitor experience.